### PR TITLE
Update mock socket

### PR DESCRIPTION
**Pull request by tomalec**
Update thoov/mock-socket@8.0.5 (from 6.0.4)

### CODE REVIEW

File: test/specs/palindrom.errors.test.js [hunk 1]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
...
-                        1}, variable path is: /val`
-                );
             });
-            it('Outgoing socket patch: out of range numbers should call onOutgoingPatchValidationError event with a RangeError', async () => {
-                const server = new MockSocketServer(getTestURL('testURL', false, true));
-
-                fetchMock.mock(getTestURL('testURL'), {
-                    status: 200,
-                    body: '{"val": 100}'
+            context('WebSocker', () => {
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: `WebSocker` -> `WebSocket`

File: test/specs/palindrom.errors.test.js [hunk 2]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
...
-                        obj.val = Number.MAX_SAFE_INTEGER + 1
-                    ,
-                    onOutgoingPatchValidationError: spy
+                it('Outgoing patch: out of range numbers should call onOutgoingPatchValidationError event with a RangeError', async () => {
+                    const onOutgoingPatchValidationError = sinon.spy().named('onOutgoingPatchValidationError');
+    
+                    new Palindrom({
+                        remoteUrl,
+                        onStateReset: obj =>
+                            obj.val = Number.MAX_SAFE_INTEGER + 1
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: `obj.val` -> `obj.value`

File: test/specs/palindrom.errors.test.js [hunk 3]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
...
+                    assert(onOutgoingPatchValidationError.calledOnce, `Expected \`onOutgoingPatchValidationError\` to be called once, but was called ${onOutgoingPatchValidationError.callCount} times`);
+    
+                    const errorPassed = onOutgoingPatchValidationError.getCall(0).args[0];
+    
+                    assert(errorPassed instanceof RangeError);
+    
+                    assert.equal(
+                        errorPassed.message,
+                        `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: ${Number.MAX_SAFE_INTEGER +
+                            1}, variable path is: /val`
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: `obj.val` -> `obj.value`

File: test/specs/palindrom.errors.test.js [hunk 4]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
...
-                    status: 200,
-                    body: '{"val": 100}'
+                it('Outgoing patch: out of range numbers should call onOutgoingPatchValidationError event with a RangeError', async () => {    
+                    const onOutgoingPatchValidationError = sinon.spy().named('onOutgoingPatchValidationError');
+    
+                    const palindrom = new Palindrom({
+                        remoteUrl,
+                        useWebSocket: true,
+                        onOutgoingPatchValidationError,
+                        onStateReset: obj => obj.val = Number.MAX_SAFE_INTEGER + 1,
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: `obj.val` -> `obj.value`

File: test/specs/palindrom.errors.test.js [hunk 5]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
...
+    
+                    assert(onOutgoingPatchValidationError.calledOnce);
+    
+                    const errorPassed = onOutgoingPatchValidationError.getCall(0).args[0];
+    
+                    assert(errorPassed instanceof RangeError);
+    
+                    assert.equal(
+                        errorPassed.message,
+                        `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in a patch, value is: ${Number.MAX_SAFE_INTEGER + 1}, variable path is: /val`
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: `obj.val` -> `obj.value`

File: test/specs/palindrom.errors.test.js [hunk 6]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
...
+                it('Incoming patch: out of range numbers should call onIncomingPatchValidationError event with a RangeError', async () => {    
+                    const onIncomingPatchValidationError = sinon.spy().named('onIncomingPatchValidationError');
+
+                    mockSocketServer.on('connection', socket => {
+                        /* prepare response */
+                        socket.on('message', function(){
+                            // socketMessageSpy(...arguments);
+                            /* respond */
+                            socket.send(
+                                `[{"op": "replace", "path": "/val", "value": ${Number.MAX_SAFE_INTEGER +
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: `obj.val` -> `obj.value`

File: test/specs/palindrom.errors.test.js [hunk 7]
```diff
@@ -211,122 +211,125 @@ describe('Palindrom', () => {
             });
         });
         context('Numbers Validation', function() {
-            it('Initial HTTP response: out of range numbers should call onIncomingPatchValidationError event with a RangeError', async () => {
-                fetchMock.mock(getTestURL('testURL'), {
+            const remoteUrl = getTestURL('testURL');
+            beforeEach(() => {
+                fetchMock.mock(remoteUrl, {
                     status: 200,
                     body: `{"value": ${Number.MAX_SAFE_INTEGER + 1}}`
```
Comment (warpech) [inline test/specs/palindrom.errors.test.js]: Moving mock server declaration to `beforeEach` caused some confusion between properties `val` and `value` used in some tests. Many tests still do replace operations on `/val`, simply using the fact that replace acts like add if a property does not exist. I find it not clean.

Maybe let's rename whatever is called `value` and `val` to something more distinct, like `amount`

**Comment by tomalec** [comment]
All addressed, I also added more time for waiting for HTTP establish to hopefully reduce some tests' flakiness.

**Pull request merged**